test/object-store/sync/flx_sync.cpp

```diff
@@ -3299,6 +3300,79 @@ TEST_CASE("flx: bootstrap changesets are applied continuously", "[sync][flx][app
     CHECK(user_commit_version == bootstrap_version + 1);
 }
 
+TEST_CASE("flx: async open + register subscription callack while bootstrapping", "[sync][flx][async-open]") {
+    std::mutex mutex_task_completed;
+    std::condition_variable cv;
```

Comment:
nit: you could use promises instead of condition_variable

Reply:
Indeed, but once I call `future.get()`, I can't call it again to wait on the same promise, I need to reset the future and the promise in order to wait a second time for the same callback. It seems the CV is better for this type of test.